api/tests/Feature/AssessmentResultTest.php

```diff
@@ -0,0 +1,387 @@
+
+use function PHPUnit\Framework\assertNull;
+
+class AssessmentResultTest extends TestCase
+{
+    use MakesGraphQLRequests;
+    use RefreshDatabase;
+    use RefreshesSchemaCache;
+
+    protected $teamUser;
+
+    protected $team;
+
+    protected $pool;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->seed(RolePermissionSeeder::class);
+        $this->bootRefreshesSchemaCache();
+        $this->team = Team::factory()->create();
+        $this->pool = Pool::factory()->draft()->create([
+            'team_id' => $this->team->id,
+        ]);
+        $this->teamUser = User::factory()
+            ->asApplicant()
+            ->asPoolOperator($this->team->name)
+            ->create([
+                'email' => 'team-user@test.com',
+                'sub' => 'team-user@test.com',
+            ]);
+    }
+
+    protected $createAssessmentResult =
+    /** @lang GraphQL */
+    '
+        mutation createAssessmentResult($createAssessmentResult: CreateAssessmentResultInput!){
+            createAssessmentResult(createAssessmentResult: $createAssessmentResult) {
+                id
+                assessmentStep {
+                    id
+                }
+                poolCandidate {
+                    id
+                }
+                poolSkill {
+                    id
+                }
+                assessmentResultType
+            }
+        }
+    ';
+
+    protected $updateAssessmentResult =
+    /** @lang GraphQL */
+    '
+        mutation updateAssessmentResult($updateAssessmentResult: UpdateAssessmentResultInput!){
+            updateAssessmentResult(updateAssessmentResult: $updateAssessmentResult) {
+                id
+                assessmentResultType
+                assessmentDecision
+                justifications
+            }
+        }
+    ';
+
+    protected $deleteAssessmentResult =
+    /** @lang GraphQL */
+    '
+        mutation deleteAssessmentResult($id: UUID!){
+            deleteAssessmentResult(id: $id) {
+                id
+            }
+        }
+    ';
+
+    // test creating assessment results
+    public function testCreatingAssessmentResult(): void
+    {
+        $skill = Skill::factory()->create();
+        $parentAssessmentStep = AssessmentStep::factory()->create([
+            'pool_id' => $this->pool->id,
+        ]);
+        $parentPoolCandidate = PoolCandidate::factory()->create([
+            'pool_id' => $this->pool->id,
+        ]);
+        DB::table('pool_skill')->insert([
+            'pool_id' => $this->pool->id,
+            'skill_id' => $skill->id,
+            'type' => PoolSkillType::ESSENTIAL->name,
+        ]);
+        $poolSkillModel = PoolSkill::first();
+
+        // can create assessment result without pool skill
+        $this->actingAs($this->teamUser, 'api')
+            ->graphQL(
+                $this->createAssessmentResult,
+                [
+                    'createAssessmentResult' => [
```

Comment:
Wait, can you create SKILL-type assessment results which aren't linked to any skills? Maybe we _shouldn't_ allow that 🤔

Reply:
Hm, given the exception I think is pretty explicitly intended for education only. I think it is entirely reasonable to incorporate that in validation. 
Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8302/commits/85a412c79adc3ec89fe1d97c722ff1d2183b2fd8